Source/ProjectNL/GAS/Ability/Active/Default/Targeting/AT_TargetingEnemy.cpp

```diff
@@ -15,66 +15,51 @@
 UAT_TargetingEnemy* UAT_TargetingEnemy::InitialEvent(UGameplayAbility* OwningAbility)
 {
 	UAT_TargetingEnemy* NewEvent = NewAbilityTask<UAT_TargetingEnemy>(
-	OwningAbility);
-	
+		OwningAbility);
+
 	return NewEvent;
 }
 
 void UAT_TargetingEnemy::Activate()
 {
-	
 	Super::Activate();
-	bTickingTask =true;
+	OriginalCharacterSpeed = Cast<ACharacter>(GetAvatarActor())->GetCharacterMovement()->MaxWalkSpeed;
+	SaveSpringArmSettings(GetAvatarActor()->FindComponentByClass<UPlayerSpringArmComponent>());
+	bTickingTask = true;
 	TargetNearestEnemy();
 }
 
 
-
 void UAT_TargetingEnemy::TickTask(float DeltaTime)
 {
 	Super::TickTask(DeltaTime);
-	UE_LOG(LogTemp, Warning, TEXT("UAT_TargetingEnemy Tasking"));
-
-	
+//	UE_LOG(LogTemp, Warning, TEXT("UAT_TargetingEnemy Tasking"));
 	if (NearestEnemy && NearestEnemyCheck)
 	{
 		FGameplayTagContainer ActiveTags;
 		//활성화된 컨테이너태그를 ActiveTAgs에 저장
 		Cast<APlayerCharacter>(GetAvatarActor())->GetAbilitySystemComponent()->GetOwnedGameplayTags(ActiveTags);
-		//if (!ActiveTags.HasTag(FGameplayTag::RequestGameplayTag(FName("Character.Roll"))))
-	//	{
-			LockOnTarget(NearestEnemy);
-		//}
+
+		LockOnTarget(NearestEnemy);
+
 		CameraRotation(DeltaTime);
-		ACharacter* OwnerCharacter = Cast<ACharacter>(GetAvatarActor());
-		
 	}
 }
 
 void UAT_TargetingEnemy::TargetNearestEnemy()
 {
+	NearestEnemyCheck = (NearestEnemyCheck) ? false : true;
 	NearestEnemy = FindNearestTarget();
-	USpringArmComponent* SpringArm = GetAvatarActor()->FindComponentByClass<USpringArmComponent>();
-	APlayerController* PlayerController = Cast<
-		APlayerController>(Cast<ACharacter>(GetAvatarActor())->GetController());
-	if (NearestEnemyCheck == false)
-	{
-		PlayerController->SetIgnoreLookInput(true); // 회전 입력 비활성화
-		NearestEnemyCheck = true;
-	}
-	else
+	if (!NearestEnemyCheck)
 	{
-		NearestEnemyCheck = false;
+		ReleaseLockOnTarget(); //어빌리티를 종료시킵니다.
 	}
-	if (NearestEnemy && NearestEnemyCheck)
+	else if (NearestEnemyCheck && NearestEnemy) //클릭과 타겟 캐릭터 상대방이 둘다 있을경우
 	{
-		Cast<ACharacter>(GetAvatarActor())->GetCharacterMovement()->MaxWalkSpeed = 250.0f;
-	//	Cast<ISLInteractionInterface>(GetAvatarActor())->NotifyTargetingEnemy(true);
-	}
-	else if (!NearestEnemy && !NearestEnemyCheck)
-	{
-		ReleaseLockOnTarget();
-	//	Cast<ISLInteractionInterface>(GetAvatarActor())->NotifyTargetingEnemy(false);
+		APlayerController* PlayerController = Cast<
+			APlayerController>(Cast<ACharacter>(GetAvatarActor())->GetController());
+		PlayerController->SetIgnoreLookInput(true); // 회전 입력 비활성화
+		Cast<ACharacter>(GetAvatarActor())->GetCharacterMovement()->MaxWalkSpeed = 250.0f; //속도 감소
```

Comment:
```suggestion
		if (APlayerController* PlayerController = Cast<
			APlayerController>(Cast<ACharacter>(GetAvatarActor())->GetController())) {
		    PlayerController->SetIgnoreLookInput(true); // 회전 입력 비활성화
		    Cast<ACharacter>(GetAvatarActor())->GetCharacterMovement()->MaxWalkSpeed = 250.0f;
		}
```

Cast<ACharacter>(GetAvatarActor())->GetCharacterMovement()->MaxWalkSpeed = 250.0f; 의 경우는 제가 추후 Attribute의 값을 변경할 수 있도록 처리해두겠습니다.

Reply:
넵 알겠습니당